am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -199,27 +202,26 @@ private boolean explicitAccessType(Set<String> userRoles) {
      * @param serviceName  name of service
      * @param resourceType type of resource
      * @param resourceName name of a resource
-     * @param roleNames    A set of role names. Currently only one role name is supported but
-     *                     in future implementations we shall support having multiple role names
+     * @param userRoles    A set of user roles
      * @return a map of attributes and their corresponding permissions or null
      */
-    @SuppressWarnings("PMD") // AvoidLiteralsInIfCondition: magic number used until multiple roles are supported
-    public Map<JsonPointer, Set<Permission>> getRolePermissions(
-        @NonNull String serviceName, @NonNull String resourceType,
-        @NonNull String resourceName, @NonNull Set<String> roleNames) {
-        if (roleNames.size() > 1) {
-            throw new IllegalArgumentException("Currently a single role only is supported. "
-                + "Future implementations will allow for multiple roles.");
-        }
-
-        List<RoleBasedAccessRecord> roleBasedAccessRecords = jdbi.withExtension(AccessManagementRepository.class,
-            dao -> dao.getRolePermissions(serviceName, resourceType, resourceName, roleNames.iterator().next()));
-
-        if (roleBasedAccessRecords.isEmpty()) {
+    public Map<JsonPointer, Set<Permission>> getRolePermissions(@NonNull String serviceName,
+                                                                @NonNull String resourceType,
+                                                                @NonNull String resourceName,
+                                                                @NonNull Set<String> userRoles) {
+        List<Map<JsonPointer, Set<Permission>>> permissionsForRoles =
+            jdbi.withExtension(AccessManagementRepository.class, dao ->
+                userRoles.stream()
+                    .map(role -> dao.getRolePermissions(serviceName, resourceType, resourceName, role))
+                    .map(record -> record.stream()
```

Comment:
Again we should rename it more appropriately. It is not single record but role based access records.